### PR TITLE
Fixes #23994 - Do not update templates out of scope

### PR DESCRIPTION
**Pull request by xprazak2**
`JobTemplate` should import `TemplateTax` as well.

### CODE REVIEW

File: app/models/concerns/template_tax.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+  module ClassMethods
+    def find_without_name_collision(name)
+      from_local = find_or_initialize_by :name => name
+      from_global = unscoped.find_or_initialize_by :name => name
+      from_local.errors.add(:name, _("cannot be used, please choose another")) if from_local.new_record? && from_global.persisted?
+      from_local
+    end
+  end
+end
```
Comment (lzap) [inline app/models/concerns/template_tax.rb]: This is bad name for this concern, it should not mention template but rather name and also "Tax" is misleading, perhaps something like TaxonomyCollisionFinder. You can extract `:name` as parameter so it's generic and reusable across our app.

File: test/models/template_test.rb [hunk 2]
```diff
@@ -411,5 +413,42 @@ class TemplateTest < ActiveSupport::TestCase
...
+        @regular_template = FactoryBot.create(:provisioning_template, :name => 'regular template', :organizations => [@org])
+        @collision_template = FactoryBot.create(:provisioning_template, :name => 'collision template', :organizations => [@empty])
+        @common_template = FactoryBot.create(:provisioning_template, :name => 'common template', :organizations => [@empty, @org])
+        @before_org = Organization.current
+        Organization.current = @org
+      end
+
+      after do
+        Organization.current = @before_org
+      end
```
Comment (lzap) [inline test/models/template_test.rb]: Don't do this, we have `teardown :clear_current_taxonomies` on each test from ActiveSupport::TestCase already.

**Review by lzap — CHANGES_REQUESTED** [review]
Nasty.

**Comment by xprazak2** [comment]
I renamed the concern and made it more generic, made changes in tests

**Comment by lzap** [comment]
Test failures seem relevant.

**Comment by xprazak2** [comment]
[test foreman]
[test katello]
[test upgrade]

**Review by lzap — APPROVED** [review]
TY

**Comment by lzap** [comment]
Thank you very much for your contribution, this is now merged.

**Pull request merged**